### PR TITLE
Feat/disable-folder

### PR DESCRIPTION
**Pull request by kaitoz11**

### CODE REVIEW

File: src/Application/Folders/Commands/AddFolder/AddFolderCommandValidator.cs [hunk 1]
```diff
@@ -17,7 +17,8 @@ public AddFolderCommandValidator()
             .MaximumLength(256).WithMessage("Description cannot exceed 256 characters.");
 
         RuleFor(f => f.Capacity)
-            .GreaterThanOrEqualTo(1).WithMessage("Folder's capacity cannot be less than 1");
+            .NotEmpty().WithMessage("Folder is required.")
```
Comment (StarryFolf) [inline src/Application/Folders/Commands/AddFolder/AddFolderCommandValidator.cs:20]: This should have been "Folder's capacity is required"

File: tests/Application.Tests.Integration/Folders/Commands/DisableFolderTests.cs [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        };
+        // Act
+        var result = async () => await SendAsync(disableFolderCommand);
+        // Assert
+        await result.Should().ThrowAsync<KeyNotFoundException>()
+            .WithMessage("Folder does not exist.");
+    }
+
+    [Fact]
+    public async Task ShouldThrowKeyInvalidOperationException_WhenFolderAlreadyDisabled()
```
Comment (StarryFolf) [inline tests/Application.Tests.Integration/Folders/Commands/DisableFolderTests.cs:81]: inconsistent with the desired exception

File: tests/Application.Tests.Integration/Folders/Commands/DisableFolderTests.cs [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+            .WithMessage("Folder has already been disabled.");
+        
+        // Clean up
+        Remove(folder);
+        Remove(locker);
+        Remove(room);
+    }
+    
+    [Fact]
+    public async Task ShouldThrowKeyInvalidOperationException_WhenFolderHaveDocuments()
```
Comment (StarryFolf) [inline tests/Application.Tests.Integration/Folders/Commands/DisableFolderTests.cs:132]: KeyInvalidOperationException?

File: src/Api/Middlewares/ExceptionMiddleware.cs [hunk 4]
```diff
@@ -88,6 +89,13 @@ private async void HandleLimitExceededException(HttpContext context, Exception e
         await WriteExceptionMessageAsync(context, ex);
     }
 
+    private async void HandleInvalidOperationException(HttpContext context, Exception ex)
+    {
+        context.Response.StatusCode = StatusCodes.Status400BadRequest;
```
Comment (Vzart) [inline src/Api/Middlewares/ExceptionMiddleware.cs:94]: I thought we agree on 406

File: tests/Application.Tests.Integration/Folders/Commands/DisableFolderTests.cs (hunk 3 above)
Comment (kaitoz11) [reply tests/Application.Tests.Integration/Folders/Commands/DisableFolderTests.cs]: fixed at https://github.com/ProFile-org/ProFile-BE/pull/88/commits/079c2a2468676481a1e4f5b057a50c9268be09d6

File: tests/Application.Tests.Integration/Folders/Commands/DisableFolderTests.cs (hunk 2 above)
Comment (kaitoz11) [reply tests/Application.Tests.Integration/Folders/Commands/DisableFolderTests.cs]: fixed at https://github.com/ProFile-org/ProFile-BE/pull/88/commits/079c2a2468676481a1e4f5b057a50c9268be09d6

File: src/Application/Folders/Commands/AddFolder/AddFolderCommandValidator.cs (hunk 1 above)
Comment (kaitoz11) [reply src/Application/Folders/Commands/AddFolder/AddFolderCommandValidator.cs]: fixed at https://github.com/ProFile-org/ProFile-BE/pull/88/commits/76a4df4ca33e0acda026746dd9e57214d3d9e34f

File: src/Api/Middlewares/ExceptionMiddleware.cs (hunk 4 above)
Comment (kaitoz11) [reply src/Api/Middlewares/ExceptionMiddleware.cs]: fixed at https://github.com/ProFile-org/ProFile-BE/pull/88/commits/b90c26f82c8a10a4e8a8a02eeb438815d7991180

Comment (kaitoz11) [reply src/Api/Middlewares/ExceptionMiddleware.cs]: I changed to 409

**Review by ChienNQuang — APPROVED** [review]

**Pull request merged**